src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,51 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { getUser } from '../../api';
+
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  userId: number;
+  selectedTodoId: number;
+  onSetUserId: (value: number) => void;
+  onSelectedTodoId: (value: number) => void;
+  todos: Todo[];
+};
+
+export const TodoModal: React.FC<Props> = ({
+  userId,
+  todos,
+  selectedTodoId,
+  onSetUserId,
+  onSelectedTodoId,
+}) => {
+  const [user, setUser] = useState<User>();
+  const [loadingUser, setLoadingUser] = useState(false);
+
+  useEffect(() => {
+    const getUserFromServer = async () => {
+      const userFromServer = await getUser(userId);
+
+      setLoadingUser(true);
+      setUser(userFromServer);
+    };
+
+    getUserFromServer();
+  }, []);
+
+  const findClickedTodoBySelectedId = useCallback((todoId: number) => {
+    const findedTodo = todos.find(todo => todo.id === todoId);
+
+    return findedTodo;
+  }, []);
+
```

Comment:
You can save selected todo, not only id of it. And if you've saved whole todo, you would have access to userId also